### ISSUE TITLE
Calendar: Name is not defined for the selected year button

### ISSUE THREAD

**Issue by qqli1209** [OfficeDev/office-ui-fabric-react#10352]
### Environment Information

- **Package version(s)**: all
- **Browser and OS versions**: all

> Note that our screen reader support is limited to Edge + Narrator. Issues that do not reproduce in Edge + Narrator, and aren't caused by obvious invalid aria values, should be filed with the respective screen reading software, not the Fabric repo.

### Describe the issue:

Name is not defined for the ‘2019’ button given in calendar. Screen reader announces it as ‘2019’ button which doesn’t clear the purpose of the control.

### Please provide a reproduction of the issue in a codepen:
Go to Calendar demo page, you can see the issue.
https://developer.microsoft.com/en-us/fabric#/controls/web/calendar

#### Actual behavior:

Proper name should be defined for the button like: Selected year and screen reader should announce it as ‘Selected year ‘2019’ button’.

#### Expected behavior:

Proper name should be defined for the button like: Selected year and screen reader should announce it as ‘Selected year ‘2019’ button’. Or change the `aria-label` from '2019' to 'Selected year 2019'.

**Comment by KevinTCoughlin** [comment]
@lorejoh12 are you open to me implementing the above a11y request?

**Comment by dzearing** [comment]
Ping @lorejoh12

**Comment by dzearing** [comment]
I would assume we'd want the user to provide a localized format string.

**Comment by KevinTCoughlin** [comment]
Below is the DOM:

![image](https://user-images.githubusercontent.com/706967/66775864-dd22a700-ee79-11e9-9881-9145a98ce5e6.png)

We'll need to support user's passing in a localized string in the impacted Calendar controls for them to mitigate the a11y validation error.

**Comment by jamesedjac** [comment]
@KevinTCoughlin and @lorejoh12 can we get an update on this, we still have an open bug to resolve?

**Comment by KevinTCoughlin** [comment]
@lorejoh12 can you comment on how to address this issue based on the LoC impacted above in https://github.com/OfficeDev/office-ui-fabric-react/issues/10352#issuecomment-541858641?

**Comment by lorejoh12** [comment]
We need to add a string to the ICalendarStrings interface for "currentDateFormatString", much like the existing "weekNumberFormatString".

Then in CalendarMonth.tsx, we can have the aria-label be set as:
```
aria-label={format(currentDateFormatString, dateTimeFormatter.formatYear(navigatedDate))}
```

Two thoughts here.

"Selected date 2019" is pretty ambiguous here. We actually do have a concept of "selected date", when you actually click a day, and it highlights it blue by default. Just because you navigate the month forwards or backwards doesn't mean that selection has changed, so I'm not sure if that's the best aria label here. But if that's enough for you guys, I'm okay with it.

Secondly, do we want to make the same update to the other part of the picker? When looking at the days in a single month, the aria label is just "October 2019", same as the year picker's "2019". Do we want to update that as well to be "Selected date October 2019", or is that one clear enough?

**Comment by KevinTCoughlin** [comment]
> We need to add a string to the ICalendarStrings interface for "currentDateFormatString", much like the existing "weekNumberFormatString".
> 
> Then in CalendarMonth.tsx, we can have the aria-label be set as:
> 
> ```
> aria-label={format(currentDateFormatString, dateTimeFormatter.formatYear(navigatedDate))}
> ```
> 
> Two thoughts here.
> 
> "Selected date 2019" is pretty ambiguous here. We actually do have a concept of "selected date", when you actually click a day, and it highlights it blue by default. Just because you navigate the month forwards or backwards doesn't mean that selection has changed, so I'm not sure if that's the best aria label here. But if that's enough for you guys, I'm okay with it.
> 
> Secondly, do we want to make the same update to the other part of the picker? When looking at the days in a single month, the aria label is just "October 2019", same as the year picker's "2019". Do we want to update that as well to be "Selected date October 2019", or is that one clear enough?

@jamesedjac can you comment on the above as domain expert before we implement a fix?

**Comment by jamesedjac** [comment]
Hi @lorejoh12 and @KevinTCoughlin 

I think the core thing we are running into with the label has to do with its function as a button rather than making it clear the the year displayed is the current selection. It needs a label that makes it clear that pressing the button activates a mode that allows you to select the year. Something like "2019, select to change the year". The other controls would be fine as is, because they communicate enough about their purpose with the simple label.

Essentially, we can assume users would understand pressing the "March" button means they are selecting the month of March. The issue with the year button, is that pressing the 2019 button doesn't mean you are selecting the year 2019, so it needs a more complex label

**Comment by lorejoh12** [comment]
I think that's a reasonable label, "2019, select to change the year". Lets go with that one, using the string formatting I described above. We won't touch the month button.

**Comment by jamesedjac** [comment]
Thank you @lorejoh12. When can we expect a fix?

**Comment by lorejoh12** [comment]
For some reason the PR merging didn't auto close the issues... closing, this should be fixed now.
